time stamp
int64 x
int64 y
int32 budgetResidual
float32 variance